inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -733,45 +993,20 @@ struct jit_uni_reduce_kernel_f32 : public jit_uni_reduce_kernel, public jit_gene
             vextractf32x4(xmm_aux3, zmm_dst, 3);
             horiz_ps(xmm_aux2, xmm_aux3);
             horiz_ps(xmm_aux1, xmm_aux2);
-            load_embedded_horiz_store(xmm_aux1, dst_dt);
+            horiz_store(xmm_aux1, dst_dt, load_embedded);
         }
     }
 
-    inline void load_embedded_horiz_store(Xbyak::Xmm xmm_dst, memory::data_type dst_dt) {
+    inline void horiz_store(Xbyak::Xmm xmm_dst, memory::data_type dst_dt, bool load_embedded) {
         movshdup(xmm_aux3, xmm_dst); // dst:1,2,3,4; aux3:2,2,4,4
         horiz_ps(xmm_dst, xmm_aux3); // dst:f(1,2),f(2,2),f(3,4),f(4,4)
         movhlps(xmm_aux3, xmm_dst);  // aux3:f(3,4),f(4,4),4,4
         horiz_ps(xmm_dst, xmm_aux3); // dst:f(1,2,3,4),...
-        load_scalar(xmm_aux3, ptr[reg_dst], dst_dt);
-
-        switch (dst_dt) {
-            case memory::data_type::f32:
-            case memory::data_type::bf16:
-                horiz_ps(xmm_dst, xmm_aux3);
-                store_scalar(ptr[reg_dst], xmm_dst, dst_dt);
-                break;
-            case memory::data_type::s32:
-                horiz_ps(xmm_dst, xmm_aux3);
-                uni_vcvtps2dq(xmm_dst, xmm_dst);
-                movss(ptr[reg_dst], xmm_dst);
-                break;
-            case memory::data_type::u8:
-                horiz_ps(xmm_dst, xmm_aux3);
-                uni_vcvtps2dq(xmm_dst, xmm_dst);
-                uni_vpackusdw(xmm_dst, xmm_dst, xmm_dst);
-                uni_vpackuswb(xmm_dst, xmm_dst, xmm_dst);
-                pextrb(ptr[reg_dst], xmm_dst, 0);
-                break;
-            case memory::data_type::s8:
-                horiz_ps(xmm_dst, xmm_aux3);
-                uni_vcvtps2dq(xmm_dst, xmm_dst);
-                uni_vpackssdw(xmm_dst, xmm_dst, xmm_dst);
-                uni_vpacksswb(xmm_dst, xmm_dst, xmm_dst);
-                pextrb(ptr[reg_dst], xmm_dst, 0);
-                break;
-            default:
-                assert(!"unknown dst_dt");
+        if (load_embedded) {
+            load_scalar(xmm_aux3, ptr[reg_dst], dst_dt);
+            horiz_ps(xmm_dst, xmm_aux3);
         }
+        store_scalar(ptr[reg_dst], xmm_dst, dst_dt);
     }
 
     inline void horiz_ps(const Xmm& xmm, const Operand& op) {
```

Comment:
could you please update to uni_xxx for this function.

Reply:
Fixed. Thanks Chenhu!